web/ui/react-app/src/pages/serviceDiscovery/Services.tsx

```diff
@@ -20,100 +19,95 @@ export interface TargetLabels {
+  const labels: Record<string, TargetLabels[]> = {};
 
-    for (const target of response) {
-      const name = target.scrapePool;
-      if (!labels[name]) {
-        labels[name] = [];
-      }
-      labels[name].push({
-        discoveredLabels: target.discoveredLabels,
-        labels: target.labels,
-        isDropped: false,
-      });
+  for (const target of activeTargets) {
+    const name = target.scrapePool;
+    if (!labels[name]) {
+      labels[name] = [];
     }
+    labels[name].push({
+      discoveredLabels: target.discoveredLabels,
+      labels: target.labels,
+      isDropped: false,
+    });
+  }
 
-    for (const target of dropped) {
-      const { job: name } = target.discoveredLabels;
-      if (!labels[name]) {
-        labels[name] = [];
-      }
-      labels[name].push({
-        discoveredLabels: target.discoveredLabels,
-        isDropped: true,
-        labels: {},
-      });
+  for (const target of droppedTargets) {
+    const { job: name } = target.discoveredLabels;
+    if (!labels[name]) {
+      labels[name] = [];
     }
+    labels[name].push({
+      discoveredLabels: target.discoveredLabels,
+      isDropped: true,
+      labels: {},
+    });
+  }
 
-    return labels;
-  };
+  return labels;
+};
 
-  if (error) {
-    return (
-      <Alert color="danger">
-        <strong>Error:</strong> Error fetching Service-Discovery: {error.message}
-      </Alert>
-    );
-  } else if (response.data) {
-    const targets = processSummary(response.data);
-    const labels = processTargets(response.data.activeTargets, response.data.droppedTargets);
+export const ServiceDiscoveryContent: FC<ServiceMap> = ({ activeTargets, droppedTargets }) => {
+  const targets = processSummary(activeTargets, droppedTargets);
+  const labels = processTargets(activeTargets, droppedTargets);
 
-    return (
-      <>
-        <h2>Service Discovery</h2>
-        <ul>
-          {Object.keys(targets).map((val, i) => (
-            <li key={i}>
-              <a href={'#' + val}>
-                {' '}
-                {val} ({targets[val].active} / {targets[val].total} active targets){' '}
-              </a>
-            </li>
-          ))}
-        </ul>
-        <hr />
-        {Object.keys(labels).map((val: any, i) => {
-          const value = labels[val];
-          return <LabelsTable value={value} name={val} key={Object.keys(labels)[i]} />;
-        })}
-      </>
-    );
-  }
-  return <FontAwesomeIcon icon={faSpinner} spin />;
+  return (
+    <>
+      <h2>Service Discovery</h2>
+      <ul>
+        {mapObjEntries(targets, ([k, v]) => (
+          <li key={k}>
+            <a href={'#' + k}>
+              {k} ({v.active} / {v.total} active targets)
+            </a>
+          </li>
+        ))}
+      </ul>
+      <hr />
+      {mapObjEntries(labels, ([k, v]) => {
+        return <LabelsTable value={v} name={k} key={k} />;
+      })}
+    </>
+  );
+};
+ServiceDiscoveryContent.displayName = 'ServiceDiscoveryContent';
```

Comment:
IMO these display names should better be normal human-readable words like "service discovery information", so the error becomes "Error: Error fetching service discovery information: Failed to fetch" instead of something more cryptic.

Reply:
IMO ```displayName``` should be used as intended i.e should display the name of the component.
The default error message template use that name but if we want to provide more than that we can use ```customErrorMsg``` prop

Reply:
But the component name isn't really relevant to the user and just looks cryptic... so yeah, let's at least show the user the normal English wording :)

Reply:
Yeah, for the components which have not user-friendly names I'll create a custom error message.